### ISSUE TITLE
Script check timeout behavior

### ISSUE THREAD

**Issue by shmizad** [hashicorp/consul#1508]
Do script checks have a timeout of 30 seconds? As soon as a script takes more than 30s to run, it's marked as critical and no output is printed whatsoever.

Also I can't find any log entries that say that the script timed out.

Unfortunately I couldn't find any documentation that mentions this timeout. It would be nice to be able to set the timeout in the config file manually if a longer timeout is necessary and to have a meaningful error message.

The first script will run as expected, the other one will result in the check being marked 'critical' and the output string is empty:

```
#!/bin/sh

sleep 29
echo "you can read this"
```

```
#!/bin/sh

sleep 31
echo "you cannot read this"
```

This issue happened in 0.5.2 and 0.6.0. I didn't test it on other versions.

**Comment by slackpad** [comment]
Hi @shmizad - thanks for opening an issue! There is a 30 second limit enforced here - https://github.com/hashicorp/consul/blob/master/command/agent/check.go#L175. We'd need to look through the surrounding code to make sure it's ok to tune this with a config, but we can definitely document this and provide a good error message (it gets emitted through a channel but not logged for the timeout case).

**Comment by slackpad** [comment]
Closed via #1762.
